wgpu-core/src/lock/validating.rs

```diff
@@ -0,0 +1,91 @@
+/*! Methods that validate lock acquisition order. */
+#![allow(dead_code)]
+
+use super::LockRank;
+use std::{cell::Cell, panic::Location};
+
+pub struct Mutex<T> {
+    inner: parking_lot::Mutex<T>,
+    rank: LockRank,
+}
```

Comment:
- [x] issue: We should add some documentation to this symbol. At the very least, we should point folks towards this and the non-validating implementation being selected based on Cargo features, preferably by linking to the details you've already written elsewhere. Maybe something like this:

```suggestion
/// Mutual exclusion lock variant for resources in core which validates lock ordering. See also [`super`].
pub struct Mutex<T> {
    inner: parking_lot::Mutex<T>,
    rank: LockRank,
}
```

Reply:
- [ ] This feedback also applies to the `non_validating::Mutex` symbol, of course.

Reply:
All `Mutex` and guard types now have docs, and there's an explanation of how we select an implementation.

Reply:
All these have docs now.